core/src/main/scala/org/apache/spark/NewAccumulator.scala

```diff
@@ -197,7 +199,7 @@ private[spark] object AccumulatorContext {
    * TODO: Don't use a global map; these should be tied to a SparkContext (SPARK-13051).
    */
   @GuardedBy("AccumulatorContext")
-  val originals = new java.util.HashMap[Long, jl.ref.WeakReference[NewAccumulator[_, _]]]
+  private val originals = new java.util.HashMap[Long, jl.ref.WeakReference[NewAccumulator[_, _]]]
```

Comment:
Have you considered using ConcurrentHashMap for originals ?
The two calls in register() can then be replaced with putIfAbsent().